pytorch_lightning/callbacks/quantization.py

```diff
@@ -63,9 +66,9 @@ def wrap_quantize_forward_context(model: 'pl.LightningModule', func: Callable) -
     """
     # todo: consider using registering hook before/after forward
     @functools.wraps(func)
-    def wrapper(data) -> Any:
-        data = model.quant(data)
-        data = func(data)
+    def wrapper(*data, **kwargs) -> Any:
+        data = [q(d) for q, d in zip(model.quants, data)]
+        data = func(*data, **kwargs)
         data = model.dequant(data)
```

Comment:
Can we add multi-output support by the same occasion ?

Reply:
We can re-use a single dequant stub for multiple outputs (since it's [stateless](https://discuss.pytorch.org/t/what-do-de-quantstub-actually-do/82402/2)), so yes it would just be a matter of iterating over the outputs

Reply:
I realized if we do `data = [model.dequant(d) for d in data]`, then we must make an opinionated choice about whether the final `data` is a list, tuple, or tensor. Since we don't know the expected return type of a users model, I don't see a good way to guarantee that the wrapper is consistent with their default forward behavior. Do you think we should just pick one of the iterable types and note it in the docs?

Reply:
What about using the type of the initial `data`?

Reply:
Yes @carmocca we could check the initial type of data returned from `func`, that's a good solution. I don't think this is necessary for the `QuantStub` application prior to `func` because the `*data` invocation is going to unpack the iterable regardless of what exact type it is